src/pulsar/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::{borrow::Cow, collections::BTreeMap};
+
+use testcontainers::{
+    core::{CmdWaitFor, ContainerPort, ContainerState, ExecCommand, Mount, WaitFor},
+    Image, TestcontainersError,
+};
+
+const NAME: &str = "apachepulsar/pulsar";
+const TAG: &str = "2.10.6";
+
+const PULSAR_PORT: ContainerPort = ContainerPort::Tcp(6650);
+const ADMIN_PORT: ContainerPort = ContainerPort::Tcp(8080);
+
+/// Module to work with [`Apache Pulsar`] inside of tests.
+/// **Requires protoc to be installed, otherwise will not build.**
+///
+/// This module is based on the official [`Apache Pulsar docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{pulsar, testcontainers::runners::SyncRunner};
+///
+/// let pulsar = pulsar::Pulsar::default().start().unwrap();
+/// let http_port = pulsar.get_host_port_ipv4(6650).unwrap();
+///
+/// // do something with the running pulsar instance..
+/// ```
+///
+/// [`Apache Pulsar`]: https://github.com/apache/pulsar
+/// [`Apache Pulsar docker image`]: https://hub.docker.com/r/apachepulsar/pulsar/
+#[derive(Debug, Clone)]
+pub struct Pulsar {
+    data_mount: Mount,
+    env: BTreeMap<String, String>,
+    admin_commands: Vec<Vec<String>>,
+}
+
+impl Default for Pulsar {
+    /**
+     * Starts an in-memory instance in dev mode, with horrible token values.
+     * Obviously not to be emulated in production.
+     */
```

Comment:
```suggestion
```

a nit: I think it's excessive,  `testcontainers` are in general not indented to be used in production and default insecure values are expected for most of the containers.

And additionally: it's instantiation, without startup. User can override all tokens and etc before starting the container